src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,202 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size == 0) {
+            head = newNode;
+            tail = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        isIndexValidToAddNode(index);
+        Node<T> oldNode = findByIndex(index);
+        Node<T> newNode = new Node<>(null, value, null);
+
+        if (index == 0 && size != 0) {
+            head.prev = newNode;
+            newNode.next = head;
+            head = newNode;
+            size++;
+        } else if (index == 0) {
+            add(value);
+        } else if (index == size) {
```

Comment:
Complicated method, a lot of if else statements introduce some methods to increase readablility